wire-runtime/src/test/proto-kotlin/com/squareup/wire/protos/kotlin/person/Person.kt

```diff
@@ -21,7 +24,18 @@ data class Person(
     val email: String? = null,
     val phone: List<PhoneNumber> = emptyList(),
     val unknownFields: ByteString = ByteString.EMPTY
-) {
+) : Message<Person, Person.Builder>(ADAPTER, unknownFields) {
+    @Deprecated(
+            message = "Not relevant for data classes, use copy()",
```

Comment:
I don't find this to be true since there's no way to have multiple functions contribute to the building of one object.

Reply:
Also this is going to cause deprecations on the Java side for all these things where Builder is the only thing you have

Reply:
The idea is to generate different code for Kotlin-only/mixed codebases: for Kotlin-only, we don't really need Builders, but we still have to extend `Message` and provide a `Message.Builder` implementation, deprecation annotations should prevent people from using the Builder and prefer `copy()`; for mixed codebases we'll generate full versions of Builders to remain source-compatible with Java protos - this will be hidden behind a compiler flag.

Reply:
In that case they should be marked as HIDDEN without a ReplaceWith

Reply:
Is this `HIDDEN` the solution to https://youtrack.jetbrains.com/issue/KT-24769 ?

Reply:
Can kotlin functions with HIDDEN annotations be accessed from JAVA?